src/test/rustdoc-ui/intra-doc/disambiguator-mismatch.stderr

```diff
@@ -2,94 +2,138 @@ error: incompatible link kind for `S`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `T`
   --> $DIR/disambiguator-mismatch.rs:34:14
    |
 LL | /// Link to [struct@T]
-   |              ^^^^^^^^ help: to link to the trait, prefix with `trait@`: `trait@T`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a trait, which is not a struct
+help: to link to the trait, prefix with `trait@`
+   |
+LL | /// Link to [trait@T]
+   |              ^^^^^^
 
 error: incompatible link kind for `m`
   --> $DIR/disambiguator-mismatch.rs:39:14
    |
 LL | /// Link to [derive@m]
-   |              ^^^^^^^^ help: to link to the macro, add an exclamation mark: `m!`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a macro, which is not a derive macro
+help: to link to the macro, add an exclamation mark
+   |
+LL | /// Link to [derive@m!]
+   |                      ^
 
 error: incompatible link kind for `s`
   --> $DIR/disambiguator-mismatch.rs:44:14
    |
 LL | /// Link to [const@s]
-   |              ^^^^^^^ help: to link to the static, prefix with `static@`: `static@s`
+   |              ^^^^^^^
    |
    = note: this link resolved to a static, which is not a constant
+help: to link to the static, prefix with `static@`
+   |
+LL | /// Link to [static@s]
+   |              ^^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:49:14
    |
 LL | /// Link to [static@c]
-   |              ^^^^^^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a constant, which is not a static
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c]
+   |              ^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:54:14
    |
 LL | /// Link to [fn@c]
-   |              ^^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^^
    |
    = note: this link resolved to a constant, which is not a function
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c]
+   |              ^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:59:14
    |
 LL | /// Link to [c()]
-   |              ^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^
    |
    = note: this link resolved to a constant, which is not a function
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c()]
+   |              ^^^^^^
 
 error: incompatible link kind for `f`
   --> $DIR/disambiguator-mismatch.rs:64:14
    |
 LL | /// Link to [const@f]
-   |              ^^^^^^^ help: to link to the function, add parentheses: `f()`
+   |              ^^^^^^^
    |
    = note: this link resolved to a function, which is not a constant
+help: to link to the function, add parentheses
+   |
+LL | /// Link to [const@f()]
+   |                     ^^
```

Comment:
Hmm, this doesn't look right, it's suggesting that you include _both_ disambiguators. It should remove the prefix if it suggests a suffix.